### DISCUSSION TITLE
Astroid Captcha Library

### DISCUSSION THREAD

**Discussion by sonvnn** [templaza/astroid-framework#912] [Ideas]
Hi Astroiders,

I am happy to announce Astroid Captcha plugin. With this library you can protect your Form Builder widget as well as help verify new Joomla member registration form.

- To try it you can download v3-branch and test it on your site.
- To install it on Joomla Register form. Com Users -> Options -> User Options -> Captcha: Select "Astroid Captcha"
- To configure Astroid Captcha: Plugins -> Astroid Captcha

Some screenshots below:
![Screen Shot 2024-12-12 at 14 53 12](https://github.com/user-attachments/assets/c27d0676-a7f3-4ffa-9d4f-f1399b8f8e99)
![Screen Shot 2024-12-12 at 14 53 39](https://github.com/user-attachments/assets/89a287f4-bfdc-433e-9637-88d9d739d0b9)
![Screen Shot 2024-12-12 at 14 53 59](https://github.com/user-attachments/assets/87d323e4-35b1-4458-8cfa-196d3af9e96f)
![Screen Shot 2024-12-12 at 14 55 18](https://github.com/user-attachments/assets/ee0c2af6-1162-4514-ad25-1081119b14b4)
![Screen Shot 2024-12-12 at 14 55 53](https://github.com/user-attachments/assets/32bf52dd-cd0f-4192-9f33-baaf59c322ac)
![Screen Shot 2024-12-12 at 14 54 48](https://github.com/user-attachments/assets/a1a43fbc-4908-4953-a010-7974be9a2254)

Best regards,
Sonny

**Comment by patrickus33** [comment]
Google and GDPR are not really friends ....
May be a simple honeypot, like in Convert Form ? Or a turnstile captcha who is GDPR compatible.

**Reply by benyrfg** [reply]
I also prefer to use Turnstile, which is privacy friendly. There is an existing plugin: https://github.com/SharkyKZ/joomla-turnstile-plugin

**Reply by sonvnn** [reply]
Thank you for sharing idea. I will integrate Cloudflare Turnstile into Astroid Captcha tomorrow

**Comment by Banuk1616** [comment]
I use FlexiContact Plus from lesarbresdesign for my extensive forms. There is a nice image captcha in which you can set the number of images variably. I personally find this a great option, so far there has been no spam via these forms and everything is GDPR compliant. With the Cloudflare Turnstile anti-spam plugin, I find it extremely annoying that you have to create an account with Cloudflare. (Register at https://www.cloudflare.com/ to get your site and secret keys, and setup domains.)
Best Regards

**Comment by salahjaafar** [comment]
@sonvnn I want to commend the Astroid team for adding the Captcha plugin to enhance form security. However, to further improve this solution, I propose integrating a honeypot method.

**Comment by rinka88** [comment]
and probably in the descriptions of each Captcha Type you need to add a link to the settings page or get the secret key for the captcha

**Comment by sonvnn** [comment]
@benyrfg @Banuk1616 @patrickus33 I added CloudFlare Turnstile captcha on https://github.com/templaza/astroid-framework/commit/5d33b27463028588c409e9f322c05e460358867f
You can download RC2 and test it. https://github.com/templaza/astroid-framework/releases/tag/v3.1.9-rc2
![Screen Shot 2024-12-13 at 15 18 23](https://github.com/user-attachments/assets/93712f59-bbf3-4f95-b634-4a24d8fb2f63)
![Screen Shot 2024-12-13 at 15 18 51](https://github.com/user-attachments/assets/cd0df6d5-55a4-4881-aae5-790777728a39)

**Comment by rinka88** [comment]
in the Astroid plugin settings there is a duplicate section
![IMG_20241213_141430_133](https://github.com/user-attachments/assets/ad55b6be-47b3-4e7b-bf90-b1028489918b)

**Reply by sonvnn** [reply]
I moved them to Astroid Captcha already. Let's download rc2 and test it
